### PR TITLE
cephfs: refactor run_as_root() to user run_as_user()

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Fixes: https://tracker.ceph.com/issues/46057

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
In context to https://github.com/ceph/ceph/pull/34595/files#r415952326

File: qa/tasks/cephfs/mount.py [hunk 1]
```diff
@@ -536,16 +536,9 @@ def run_as_user(self, args, user, wait=True, stdin=None,
 
     def run_as_root(self, args, wait=True, stdin=None, check_status=True,
```
Comment (batrick) [inline qa/tasks/cephfs/mount.py:537]:
````
def run_as_root(self, **kwargs):
   return self.run_as_user(**kwargs)
````

? (Although you might need to add the stdout/stderr below.)

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/mount.py:537]: That would be certainly be better. `stdout` and `stderr` can be added by `run_as_user()`, we'll just need to do `kwargs['user'] = 'root'`.

**Comment by rishabh-d-dave** [comment]
@jtlayton @batrick I created a tracker ticket for the second commit and added Fixes line to it and the PR for the sake of bookkeeping.

**Comment by batrick** [comment]
jenkins test dashboard backend

**Comment by batrick** [comment]
http://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200623.022248

Failures unrelated.

**Pull request merged**